src/main/java/org/apache/mesos/api/JettyApiServer.java

```diff
@@ -0,0 +1,55 @@
+package org.apache.mesos.api;
+
+import org.apache.mesos.scheduler.plan.api.PlanResource;
+import org.apache.mesos.scheduler.recovery.api.RecoveryResource;
+import org.apache.mesos.state.api.StateResource;
+import org.eclipse.jetty.server.Server;
+import org.glassfish.jersey.jetty.JettyHttpContainerFactory;
+import org.glassfish.jersey.server.ResourceConfig;
+
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * A JettyApiServer takes a list of POJO JAX-RS Resources and serves them on the indicated port of 0.0.0.0.
+ */
+public class JettyApiServer {
+    private final Server server;
+
+    public JettyApiServer(
+            int port,
+            PlanResource planResource,
+            RecoveryResource recoveryResource,
+            StateResource stateResource) {
+        this(port, Arrays.asList(planResource, recoveryResource, stateResource));
+    }
+
+    public JettyApiServer(int port, Collection<Object> resources) {
+        ResourceConfig resourceConfig = new ResourceConfig();
+        for (Object resource : resources) {
+            resourceConfig.register(resource);
+        }
+
+        URI baseUri = UriBuilder
+                .fromUri("http://0.0.0.0/")
```

Comment:
Only `http` is necessary to expose the endpoints through admin router.  You would prefer `https` as well?  Investigation seems to indicate this would be non-trivial, and I don't understand the benefit.  These endpoints are by design publicly available through the admin router.  Can you clarify?

You would prefer to have the ip address configurable?